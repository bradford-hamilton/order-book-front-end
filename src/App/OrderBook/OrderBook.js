import React, { Fragment } from 'react';
import { RingLoader } from 'react-spinners';

const OrderBook = ({ type, loading, data }) => (
  <Fragment>
    <h4 className="orderBookType">
      <span className="bold">Live Order Book</span> - {type}
    </h4>
    <div className="orderColumns">
      <p>{type}</p>
      <p>Amount</p>
      <p>Exchange</p>
    </div>
    {loading ? (
      <div className="ringLoader">
        <RingLoader
          color={'#123ABC'}
          loading={true}
        />
      </div>
    ) : (
      <div className="bookBackground">
        {data.map((order, i) => (
          <div className="orderWrapper" key={i}>
            <p className={type.toLowerCase()}>{order.rate.toFixed(8)}</p>
            <p className="quantity">{order.quantity.toFixed(8)}</p>
            <p className="exchange">{order.exchange}</p>
          </div>
        ))}
      </div>
    )}
  </Fragment>
);

export default OrderBook;
