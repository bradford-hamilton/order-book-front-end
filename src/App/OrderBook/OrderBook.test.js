import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import OrderBook from './OrderBook';

configure({ adapter: new Adapter() });

function setup() {
  const props = {
    type: 'Bid',
    loading: false,
    data: [
      {
        rate: 3,
        quantity: 100,
        exchange: 'bittrex',
      },
      {
        rate: 3,
        quantity: 100,
        exchange: 'poloniex',
      },
      {
        rate: 3,
        quantity: 100,
        exchange: 'bittrex',
      },
    ],
  };
  const comp = shallow(<OrderBook {...props} />);

  return { comp, props };
}

describe('<OrderBook />', () => {
  it('renders without crashing', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<OrderBook {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
