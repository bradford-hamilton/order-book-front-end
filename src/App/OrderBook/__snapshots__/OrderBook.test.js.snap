// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderBook /> snapshot 1`] = `
Array [
  <h4
    className="orderBookType"
>
    <span
        className="bold"
    >
        Live Order Book
    </span>
     - 
    Bid
</h4>,
  <div
    className="orderColumns"
>
    <p>
        Bid
    </p>
    <p>
        Amount
    </p>
    <p>
        Exchange
    </p>
</div>,
  <div
    className="bookBackground"
>
    <div
        className="orderWrapper"
    >
        <p
            className="bid"
        >
            3.00000000
        </p>
        <p
            className="quantity"
        >
            100.00000000
        </p>
        <p
            className="exchange"
        >
            bittrex
        </p>
    </div>
    <div
        className="orderWrapper"
    >
        <p
            className="bid"
        >
            3.00000000
        </p>
        <p
            className="quantity"
        >
            100.00000000
        </p>
        <p
            className="exchange"
        >
            poloniex
        </p>
    </div>
    <div
        className="orderWrapper"
    >
        <p
            className="bid"
        >
            3.00000000
        </p>
        <p
            className="quantity"
        >
            100.00000000
        </p>
        <p
            className="exchange"
        >
            bittrex
        </p>
    </div>
</div>,
]
`;
